pkg/controller/deployment/deployment_controller.go

```diff
@@ -556,6 +557,44 @@ func (dc *DeploymentController) getPodMapForDeployment(d *extensions.Deployment,
 	return podMap, nil
 }
 
+// syncContainerImages will determine whether the deployment should be paused when any of the containers
+// has empty (" ") image field. In case empty image fields are found, this function return 'true'.
+func (dc *DeploymentController) syncContainerImages(d *extensions.Deployment) bool {
+	pendingContainerNames := util.GetEmptyContainerImages(d)
+	hasEmptyImage := len(pendingContainerNames) != 0
+
+	cond := util.GetDeploymentCondition(d.Status, extensions.DeploymentProgressing)
+	isWaiting := cond != nil && cond.Status == v1.ConditionFalse && cond.Reason == util.WaitingForImagesReason
+
+	if hasEmptyImage {
+		// Only update deployment status once.
+		if !isWaiting {
+			util.RemoveDeploymentCondition(&d.Status, extensions.DeploymentProgressing)
+			condition := util.NewDeploymentCondition(extensions.DeploymentProgressing, v1.ConditionFalse, util.WaitingForImagesReason,
+				fmt.Sprintf("waiting for container non-empty image fields (%s)", strings.Join(pendingContainerNames, ",")))
+			util.SetDeploymentCondition(&d.Status, *condition)
+			// Only fire event once and only when the update succeed.
+			if _, err := dc.client.ExtensionsV1beta1().Deployments(d.Namespace).UpdateStatus(d); err == nil {
+				dc.eventRecorder.Eventf(d, v1.EventTypeNormal, util.WaitingForImagesReason,
+					"A non-empty image field is required for containers: %s.", strings.Join(pendingContainerNames, ","))
+			}
+		}
+		return true
+	}
+	if isWaiting {
+		util.RemoveDeploymentCondition(&d.Status, extensions.DeploymentProgressing)
+		// If error occurs during updating status, don't emit the event but reconcile one more time.
+		if _, err := dc.client.ExtensionsV1beta1().Deployments(d.Namespace).UpdateStatus(d); err == nil {
+			// Only fire event once and only when the update succeed.
+			dc.eventRecorder.Eventf(d, v1.EventTypeNormal, util.WaitingForImagesReason, "This deployment has now all image fields set.")
+		}
+		// At this point we already have the images available, but let make controller reconcile this again
```

Comment:
Shouldn't you force a resync until the cached object is updated?

Reply:
Oh, well, once the cached object is updated we get an event, right?

Reply:
right

Reply:
(also this rely on the condition/reason to be set, so unless that is not reflected in cache, this will keep retrying updating the condition. once it succeed it will emit the event once).